### PR TITLE
docker: Add another pre-build.conf example

### PR DESCRIPTION
**Pull request by doanac**
This comes up every now and again

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Review by angolini — APPROVED** [review]

File: source/reference-manual/docker/containers.rst [hunk 1]
```diff
@@ -123,10 +123,29 @@ In rare occasions a Factory may need some custom code to run *before* the
 docker build logic is called on each container. This can be done with a file
 in the top-level directory of containers.git, ``pre-build.conf``.
 
-Example
-~~~~~~~
+Examples
+~~~~~~~~
+
+Here are some examples of things that can be done inside
+``pre-build.conf``.
```
Comment (kprosise) [inline source/reference-manual/docker/containers.rst:130]:
```suggestion
``pre-build.conf``:
```

File: source/reference-manual/docker/containers.rst [hunk 2]
```diff
@@ -123,10 +123,29 @@ In rare occasions a Factory may need some custom code to run *before* the
...
+
 ::
```
Comment (kprosise) [inline source/reference-manual/docker/containers.rst:132]:
```suggestion
.. code-block:: bash
```

File: source/reference-manual/docker/containers.rst [hunk 3]
```diff
@@ -123,10 +123,29 @@ In rare occasions a Factory may need some custom code to run *before* the
...
 
- # pre-build.conf
- # Create a file with build environment for container "shellhttpd":
+ # Create a file with build environment for container "shellhttpd"
```
Comment (kprosise) [inline source/reference-manual/docker/containers.rst:134]:
```suggestion
 # Create a file with build environment for container "shellhttpd":
```

File: source/reference-manual/docker/containers.rst [hunk 4]
```diff
@@ -123,10 +123,29 @@ In rare occasions a Factory may need some custom code to run *before* the
...
+ # First: Make our images build in a predictable order.
+ # This ensures 0base is built first so other containers can inherit it:
+ export IMAGES=$(find ./ -mindepth 2 -maxdepth 2 -name Dockerfile | cut -d / -f2 | sort)
+
+  # Second: Modify each container to use the locally build arch-specific base image:
+  _base_img="hub.foundries.io/${FACTORY}/0base:$(git log -1 --format=%h)-$ARCH"
+  for x in $IMAGES ; do
+      echo "Prebuild checking $x for FROM override"
+      sed -i "s|hub.foundries.io/${FACTORY}/0base|${_base_img}|" $x/Dockerfile
+  done
```
Comment (kprosise) [inline source/reference-manual/docker/containers.rst:150]: I noticed the indentation is shifted one to the right for this entire block. Is this intended?

**Review by kprosise — COMMENTED** [review]
minor suggestions, feel free to ignore them, otherwise it all lgtm.

File: source/reference-manual/docker/containers.rst (hunk 4 above)
Comment (doanac) [reply source/reference-manual/docker/containers.rst]: mistake. I force pushed this branch rebased on main, and fixed this with [8287c95](https://github.com/foundriesio/docs/pull/342/commits/8287c9596b7c13f592fb6fb699177848bfe07e16)

**Pull request merged**